app/clustercoordinators/clustercoordinator.go

```diff
@@ -0,0 +1,58 @@
+//go:generate mockgen -destination=../../mocks/app/clustercoordinators/mock_clustercoordinator.go -package mock_clustercoordinator github.com/rudderlabs/rudder-server/app/clustercoordinators ClusterManagerFactory,ClusterManager
```

Comment:
I don't think we need this here, since it is already implemented in terms of state providers in dynamicCluster manager: